### PR TITLE
feat: allow proxy auto start and stop

### PR DESCRIPTION
**Pull request by sebclrsn**
This implements auto start & stop capabilities for the ProxyAuxiliary:
- when the first CCProxy is opened and the ProxyAuxiliary isn't started yet, start it
- when the last CCProxy is closed and the ProxyAuxiliary is running, stop it
- if all auxiliaries are marked with `auto_start: False` in the config, apply this to the created ProxyAuxiliary as well

-> Goal is to entirely hide the proxy from the user as he won't be able to access it anymore in case of an automatic proxy creation

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 repos:
 -   repo: https://github.com/PyCQA/isort.git
-    rev: 5.8.0
+    rev: 5.12.0
```
Comment (sebastianpfischer) [inline .pre-commit-config.yaml:3]: I think the docs also need some update no?

File: src/pykiso/lib/auxiliaries/proxy_auxiliary.py [hunk 2]
```diff
@@ -92,9 +92,42 @@ def __init__(
...
+            communication channel closed.
+            When the first connection is opened and the counter is increased
+            from 0 to 1, the ``ProxyAuxiliary`` will be started and the attached
+            'physical' communication channel opened.
+        """
+        with self.lock:
+            return self._open_count
+
+    @_open_connections.setter
+    def _open_connections(self, value: int):
```
Comment (sebastianpfischer) [inline src/pykiso/lib/auxiliaries/proxy_auxiliary.py:120]: Would a counting-semaphore not do the trick?

**Review by sebastianpfischer — CHANGES_REQUESTED** [review]

**Review by Pog3k — APPROVED** [review]

File: src/pykiso/lib/auxiliaries/proxy_auxiliary.py (hunk 2 above)
Comment (sebclrsn) [reply src/pykiso/lib/auxiliaries/proxy_auxiliary.py:120]: Checking when `proxy.open_connections.acquire(blocking=False)` returns False seems way more obscure to me than having an explicit counter and some boolean operations

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sebclrsn) [reply .pre-commit-config.yaml:3]: Clarified - no need to specify the pre-commit hook versions in the docs :)

**Review by sebastianpfischer — APPROVED** [review]

**Pull request merged**